pkg/apis/audit/types.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package audit
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/authentication"
+)
+
+// Level defines the amount of information logged during auditing
+type Level string
+
+// Valid audit levels
+const (
+	// LevelNone disables auditing
+	LevelNone Level = "None"
+	// LevelMetadata provides the basic level of auditing.
+	LevelMetadata Level = "Metadata"
+	// LevelRequestObject provides Metadata level of auditing, and additionally
+	// logs the request object (does not apply for non-resource requests).
+	LevelRequestObject Level = "RequestObject"
+	// LevelResponseObject provides RequestObject level of auditing, and additionally
+	// logs the response object (does not apply for non-resource requests).
+	LevelResponseObject Level = "ResponseObject"
```

Comment:
See the PR description for an explanation. To elaborate a bit more on serialized data vs. runtime Objects:

My concern with serialized data is that it's protocol dependent. For instance, I'm assuming we don't want to log binary format requests, so if we get a protobuffer request, we would need to deserialize it, and then reserialize into json. Why not just include the deserialized version?